src/pq_flash_index.cpp

```diff
@@ -304,11 +313,26 @@ void PQFlashIndex<T, LabelT>::cache_bfs_levels(uint64_t num_nodes_to_cache, std:
     cur_level = std::make_unique<tsl::robin_set<uint32_t>>();
     prev_level = std::make_unique<tsl::robin_set<uint32_t>>();
 
-    for (uint64_t miter = 0; miter < num_medoids; miter++)
+    for (uint64_t miter = 0; miter < num_medoids && cur_level->size() < num_nodes_to_cache; miter++)
```

Comment:
Do you know why "cache_bfs_levels" is commented in https://github.com/microsoft/DiskANN/blob/0acff41968fa557b1099c056c78d8898326c1003/apps/search_disk_index.cpp#L123 ?

Reply:
Is it recommended to use this method? Because it is commented here, so I also do not use this method in our repo

Reply:
if we have sample query file, it is better to do caching based on most frequently visited nodes during search for sample queries, so we do that. if not, we can simply cache according to bfs from medoid(s). Both can be used, but former is usually better